design-editor/src/system/assistant/assistant-code-generator.js

```diff
@@ -0,0 +1,282 @@
+     * @param {string} [keyword=closet-component] given keyword
+     * @returns {string}
+     * @private
+     */
+    _generateId(keyword = 'closet-component') {
+        return `${keyword}-${new Date().getTime()}`;
+    }
+
+    /**
+     * Add attribute id to given element if this one doesn't exists
+     * otherwise return id of given element
+     * @param  {jQueryObject} $element - given element
+     * @param  {Object} model 
+     * @returns {string} created or existed id attribute of element
+     * @private
+     */
+    _addIDIfnotExists($element, model) {
+        let elementId = $element.attr('id');
+
+        if (!elementId) {
+            elementId = this._generateId();
+            $element.attr('id', elementId);
+            model.updateAttribute($element.attr('data-id'), 'id', elementId);
+            eventEmitter.emit(EVENTS.ReplaceCodeView);
+        } else {
+            model.updateAttribute($element.attr('data-id'), 'id', elementId);
+        }
+
+        return elementId;
+    }
+
+    /**
+     * Generate JS code responsible for selector
+     * template: document.getElementById('{{id}}')
+     * @param {HTMLElement} element
+     * @returns {string} 
+     * @private
+     */
+    _generateQuerySelector(element, model) {
+        const $element = $(element),
+            option = {
+                id: this._addIDIfnotExists($element, model)
+            };
+
+        return (Mustache.render(Sentences.querySelector, option));
+    }
+
+    /**
+     * Generate JS code for querySelector assigned to variable
+     * template: var {{varName}} = document.getElementById('{{id}}');
+     * @param {string} varName
+     * @param {HTMLElement} element
+     * @returns {*}
+     * @private
+     */
+    _generateInstanceVariable(varName, element, model) {
+        const selector = this._generateQuerySelector(element, model),
+            option = {
+                varName
+            };
+
+        return `${Mustache.render(Sentences.variable, option)}${selector}`;
+    }
+
+    /**
+     * Generate code for TAU Widget
+     * template: var {{varName}} = {{constructorName}}({{targetElement}});
+     * @param {Object} options options object
+     * @returns {string|Array} Code for TAU Widget
+     * @private
+     */
+    _generateTAUWidgetVariable(options) {
+        let code = ((options.options && (Object.keys(options.options).length > 0)) 
+            ? Sentences.tauWidgetWithOptions.map(item => Mustache.render(item, options)).join(this._lineEnding)
+            : (Mustache.render(Sentences.tauWidget, options)));
+        return code.replace(/&#39;/g, '\'');
+    }
+
+    /**
+     * Generate widget options as a multiline string
+     * @param {Array} options array of options
+     * @returns {string}
+     * @private
+     */
+    _generateWidgetOptionSentence(options) {
+        return options.reduce((previous, current) => previous + `\t${current[0]}: ${(current[1])}${this._lineEnding}`, '');
```

Comment:
this line is a litttle to long, could you split it in two? Apart from that looks fine :)

Reply:
BTW is this initial value necessary?